docs/migration-guides/v4.md

```diff
@@ -0,0 +1,399 @@
+# Migrating to Tool Kit v4
```

Comment:
suggestion: helpful to have a contents list with hyperlinks to the subheadings for easier navigation through the doc